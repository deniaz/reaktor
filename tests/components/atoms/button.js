import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Button from '../../../src/components/atoms/button';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  describe('atoms', () => {
    describe('button', () => {
      it('renders correctly.', () => {
        const clickHandler = jest.fn();
        const tree = renderer.create(<Button onClick={clickHandler} text="Example" />).toJSON();

        expect(tree).toMatchSnapshot();
      });

      it('handles click correctly.', () => {
        const clickHandler = jest.fn();
        const wrapper = shallow(<Button onClick={clickHandler} text="Example" />);
        wrapper.simulate('click');
        expect(clickHandler).toHaveBeenCalledTimes(1);
      });
    });
  });
});
