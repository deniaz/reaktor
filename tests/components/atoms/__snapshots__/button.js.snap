// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components atoms button renders correctly. 1`] = `
<button
  onClick={[Function]}
>
  Example
</button>
`;
